doc/STYLEGUIDE.md

```diff
@@ -13,6 +13,48 @@ Set up [HLint](https://github.com/ndmitchell/hlint)
 in your code environment and follow its code suggestions
 as much as you can.
 
+General naming convention
+-------------------------
+
+For anything code related: modules, functions and variables, filenames etc. we
+will use the American English spelling.
+
+### Modules and directories
+
+Module and directory names should be written as nouns:
+
+```
+Translation
+Contextualization
+Desugaring
+```
+
+We should avoid prefixes in module names as much as possible, this is what sub
+directories are for, i.e. `Frontend.Contextualize` instead of
+`FrontendContextualize`.
```

Comment:
```suggestion
We should avoid prefixes in module names as much as possible, this is what sub-directories are for, i.e. `Frontend.Contextualize` instead of `FrontendContextualize`.
```